### DISCUSSION TITLE
PHPStorm and Xdebug connection Problem on macOS

### DISCUSSION THREAD

**Discussion by SuperFlySly** [ddev/ddev#6758] [Q&A]
Unfortunately I am not able to get phpstorm and xdebug running, I get the error message from PHPStorm that PHPStorm is not able to connect and that I should set the client_host but no matter what I try it does not help. I have started ddev in debug mode and I get the following output:

DDEV:
```
2024-11-21T20:22:25.567 host.docker.internal='' because no other case was discovered
```

PHPStorm:

```
Connection was not established. Probably 'xdebug.remote_host=0.0.0.0' is incorrect. Change 'xdebug.remotehost'
```

does anyone have an idea?

**Comment by stasadev** [comment]
Hi @SuperFlySly,

See https://ddev.readthedocs.io/en/stable/users/debugging-profiling/step-debugging/#troubleshooting-xdebug

If that doesn't help, provide output for `ddev debug test`

**Comment by SuperFlySly** [comment]
that's really weird

```
sly@t3play-web:/var/www/html$ telnet host.docker.internal 9003
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused
```

**Reply by stasadev** [reply]
It can be a firewall that blocks traffic.

**Reply by rfay** [reply]
Hi @SuperFlySly - If `host.docker.internal` is resolving to 127.0.0.1, nothing will ever work. 

I means that you have changed your `xdebug_ide_location` from the default (empty) in ~/.ddev/global_config.yaml. I imagine if you `ddev config global --xdebug-ide-location=""` to set it back to the default (after looking to confirm what I'm saying first or doing `ddev config global` and showing us. But I think you ended up fiddling with it. As the docs say, it is very, very rare that anybody should fiddle with it. https://ddev.readthedocs.io/en/stable/users/configuration/config/#xdebug_ide_location

**Reply by rfay** [reply]
Did you get this sorted out @SuperFlySly ? We sure want to hear.

**Reply by SuperFlySly** [reply]
@rfay I tried your suggested solution but I still get the same error, after I updated PHPStorm to version 2024.3 it worked but I still get the following message when I run a test:

```
Testing started at 18:22 ...
Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (fallback through xdebug.client_host/xdebug.client_port).
PHPUnit 10.5.38 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.12
Time: 00:00.008, Memory: 4.00 MB

OK (1 test, 4 assertions)

Process finished with exit code 0
```

**Reply by SuperFlySly** [reply]
```
sly@t3play-web:/var/www/html$ telnet host.docker.internal 9003
Trying 192.168.65.254...
Connection failed: Connection refused
Trying fdc4:f303:9324::254...
telnet: Unable to connect to remote host: Network is unreachable
```

**Reply by rfay** [reply]
Now you're making progress. It couldn't possibly work before. It looks like either your IDE isn't listening, or you have a firewall problem. 

As @stasadev said before, please go carefully through the troubleshooting steps at https://ddev.readthedocs.io/en/stable/users/debugging-profiling/step-debugging/#troubleshooting-xdebug now.

Please make sure you have deleted any PHP configuration you altered (remove your project .ddev/php directory)

And if you provide the file output by `ddev debug test` we can help more.
